doc/releases/release-notes-4.0.rst

```diff
@@ -321,6 +326,7 @@ Drivers and Sensors
 
   * Added proper ADC2 calibration entries in ESP32.
   * Fixed calibration scheme in ESP32-S3.
+  * STM32H7: Added support for higher sampling frequencies thanks to boost mode implementation.
```

Comment:
Maybe you could also add:
* STM32: Domain clock need to be explicitly defined if property `st,adc-clock-source = <ASYNC>` is used.